### PR TITLE
Upgrade to Angular 12 and typescript 4.3

### PR DESCRIPTION
**Pull request by OscarDeDios**
# PR Details

Upgrade to Angular 12 and Typescript 4.3

## Description

Followed Angular Update Guide
 https://update.angular.io/?l=3&v=11.0-12.0 and Typescript release notes https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-3.html

## Related Issue
https://github.com/systelab/systelab-preferences/issues/50

## Motivation and Context

The motivation is to have the latest versions of the framework and typescript in order to take advantage of the lasts improvements.

## How Has This Been Tested

It has been executed the unit tests and it has been tested the systelab-components showcase with this new version.

## Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] I have read the **CONTRIBUTING** document
- [x] My code follows the code style of this project
- [ ] My change requires a change to the documentation 
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed
- [x] A new branch needs to be created from master to evolve previous versions
- [x] Increase version in package.json following [Semantic Versioning](https://semver.org/)
- [x] Add the issue into the right [project](https://github.com/systelab/systelab-preferences/projects) with the proper status (In progress)

### CODE REVIEW

File: projects/systelab-preferences/tsconfig.lib.prod.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "extends": "./tsconfig.lib.json",
   "angularCompilerOptions": {
-    "enableIvy": false
+    "compilationMode": "partial"
```
Comment (olgapuig) [inline projects/systelab-preferences/tsconfig.lib.prod.json:4]: ok. Oscar has just showed me that the recommendation is to use "partial Ivy" which goes with this configuration. Refer to https://angular.io/guide/creating-libraries#building-libraries-with-ivy

**Review by olgapuig — COMMENTED** [review]
Thank you so much for the contribution @OscarDeDios, good job! I've just added 2 minor comments related to documentation.

File: README.md [hunk 2]
```diff
@@ -55,10 +55,18 @@ The following steps should be consider when migrating from version 5.
 ```
 2. When importing services and modules import them from systelab-preferences root. In WebStorm, replace in path:
 ```
-- from 'systelab-preferences/lib.+ 
+- from 'systelab-preferences/lib.+
 - from 'systelab-preferences';
 ```
 
-## Version 11
+## Version 8
```
[the remark this answers is not in the document]
Comment (OscarDeDios) [reply README.md]: Ok, done

File: projects/systelab-preferences/src/lib/memory-storage.service.ts [hunk 3]
```diff
@@ -51,17 +51,12 @@ export class MemoryStorageService {
...
 	}
 
-	public getInCompressFormat(): any {
-		const passToServer: any = {};
-		this.preferences.forEach((value: Object, key: string) => {
-			passToServer[key] = value;
-		});
-
-		const binaryString = compressor.deflate(JSON.stringify(passToServer), {to: 'string'});
+	public getInCompressFormat(): string {
```
[the remark this answers is not in the document]
Comment (OscarDeDios) [reply projects/systelab-preferences/src/lib/memory-storage.service.ts:54]: Ok, done

**Review by olgapuig — APPROVED** [review]

**Review by oGalceran1 — APPROVED** [review]
Thanks

**Pull request merged**